witchcraft/internal/wdebug/resource.go

```diff
@@ -50,12 +50,14 @@ func RegisterRoute(router wrouter.Router, sharedSecret refreshable.String) error
 
 func (r *debugResource) ServeHTTP(rw http.ResponseWriter, req *http.Request) error {
 	ctx := req.Context()
-	authHeader, err := httpserver.ParseBearerTokenHeader(req)
-	if err != nil {
-		return errors.WrapWithNewError(err, errors.DefaultPermissionDenied)
-	}
-	if authHeader != r.SharedSecret.CurrentString() {
-		return errors.NewPermissionDenied()
+	if sharedSecret := r.SharedSecret.CurrentString(); sharedSecret != "" {
+		token, err := httpserver.ParseBearerTokenHeader(req)
+		if err != nil {
+			return errors.WrapWithUnauthorized(err)
```

Comment:
Changed this to return a 401 rather than a 403 which is more compliant with the spec. https://www.rfc-editor.org/rfc/rfc7235#section-3.1